tests/js/core/widget/core/Spin/Spin.js

```diff
@@ -1,181 +1,352 @@
+				"Option duration of Spin is " + options.duration);
+			equal(spinWidget.option("rollHeight"), options.rollHeight,
+				"Option rollHeight of Spin is " + options.rollHeight);
+			equal(spinWidget.option("itemHeight"), options.itemHeight,
+				"Option itemHeight of Spin is " + options.itemHeight);
+			equal(spinWidget.option("momentumLevel"), options.momentumLevel,
+				"Option momentumLevel of Spin is " + options.momentumLevel);
+			equal(spinWidget.option("scaleFactor"), options.scaleFactor,
+				"Option scaleFactor of Spin is " + options.scaleFactor);
+			equal(spinWidget.option("moveFactor"), options.moveFactor,
+				"Option moveFactor of Spin is " + options.moveFactor);
+			equal(spinWidget.option("loop"), options.loop,
+				"Option loop of Spin is " + options.loop);
+			equal(spinWidget.option("labels"), options.labels,
+				"Option labels of Spin is " + options.labels);
+			equal(spinWidget.option("digits"), options.digits,
+				"Option digits of Spin is " + options.digits);
+			equal(spinWidget.option("dragTarget"), options.dragTarget,
+				"Option dragTarget of Spin is " + options.dragTarget);
+
+			spinWidget.destroy();
+		});
+
+		test("Undefined values of options Spin test", 4, function () {
+			var spin = document.getElementById("spin"),
+				options = {
+					min: undefined,
+					max: undefined,
+					duration: undefined
+				},
+				spinWidget = tau.engine.instanceWidget(spin, "core.Spin", options);
+
+			spinWidget.value(undefined);
+
+			equal(spinWidget.option("min"), 0, "Spin option min with undefined value check");
+			equal(spinWidget.option("max"), 0, "Spin option max with undefined value check");
+			equal(spinWidget.option("duration"), 0, "Spin option duration with undefined value check");
+			spinWidget.value(0);
+			equal(spinWidget.value(), 0, "Default option value of Spin is " + 0);
+
+			spinWidget.destroy();
+		});
+
+		test("Values Spin test", 4, function () {
+			var spin = document.getElementById("spin"),
+				options = {
+					min: 3,
+					max: 10,
+					loop: "disabled"
+				},
+				testValues = [0, 10, 100, 11],
+				spinWidget = tau.engine.instanceWidget(spin, "core.Spin", options);
+
+			spinWidget.value(testValues[0]);
+			equal(spinWidget.value(), testValues[0], "Spin value check (0)");
+			spinWidget.value(testValues[1]);
+			equal(spinWidget.value(), testValues[1], "Spin value check (10)");
+			spinWidget.value(testValues[2]);
+			equal(spinWidget.value(), spinWidget.option("max"), "Spin value check (100) limited to max");
+			spinWidget.option("loop", "enabled");
+			spinWidget.value(testValues[3]);
+			equal(spinWidget.value(), spinWidget.option("min"), "Spin value check (max + 1) equal min for loop enabled");
+
+			spinWidget.destroy();
+		});
+
+		test("Test of _updateItems", 3, function (assert) {
+			var widget = new Spin();
+
+			helpers.stub(widget, "_removeSelectedLayout", function () {
+				assert.ok(true, "_removeSelectedLayout has been called");
+				return false;
```

Comment:
Why do you return false only for the '_removeSelectedLayout'?

Reply:
It was mistake. Fixed.